pkg/rancher-desktop/utils/protocols.ts

```diff
@@ -77,19 +36,24 @@ export const protocolsRegistered = Latch();
  * production environments.
  */
 function setupAppProtocolHandler() {
-  // TODO: #5659 - Deprecated in Electron 25; replace with protocol.handle
-  // eslint-disable-next-line deprecation/deprecation
-  const registrationProtocol = isDevBuild ? protocol.registerHttpProtocol : protocol.registerFileProtocol;
-
-  // TODO: #5659 - Deprecated in Electron 25; replace with protocol.handle
-  // eslint-disable-next-line deprecation/deprecation
-  registrationProtocol('app', (request, callback) => {
-    const relPath = decodeURI(new URL(request.url).pathname);
-    const redirectUrl = redirectedUrl(relPath);
-    const result = getProtocolResponse(request, redirectUrl, relPath);
-
-    callback(result);
-  });
+  protocol.handle(
+    'app',
+    (request) => {
+      const relPath = decodeURI(new URL(request.url).pathname);
+      const redirectUrl = redirectedUrl(relPath);
+
+      if (isDevBuild) {
+        return net.fetch(
+          redirectUrl,
+          {
+            method:   request.method,
+            referrer: request.referrer,
+          },
```

Comment:
Now that I'm looking at it again, I have questions (that are probably caused by me doing it originally): Since this is `app://`, wouldn't we _always_ be using `GET` (since we can't read out of ASAR with any other method), and never care about the referrer (for the same reason)? That is, wouldn't the extra arguments be pointless in practice?

Reply:
I think you're correct, the extra arguments do appear to be pointless here. They look safe to remove 🙂